### PR TITLE
bliss: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/16473

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by toge — DISMISSED** [review]

**Comment by MikeUnverferth** [comment]
I'd like to approve, but I'm not really qualified to test it. This is blocking the update of GDAL to Conan 2.

File: recipes/bliss/all/conanfile.py [hunk 1]
```diff
@@ -23,73 +27,64 @@ class BlissConan(ConanFile):
...
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_exact_int == "gmp":
-            self.requires("gmp/6.2.0")
+            self.requires("gmp/6.2.1", transitive_headers=True, transitive_libs=True)
```
Comment (prince-chrismc) [inline recipes/bliss/all/conanfile.py:47]: Please stop exposing libs by default 

```suggestion
            self.requires("gmp/6.2.1", transitive_headers=True)
```

File: recipes/bliss/all/conanfile.py [hunk 2]
```diff
@@ -23,73 +27,64 @@ class BlissConan(ConanFile):
...
         elif self.options.with_exact_int == "mpir":
-            self.requires("mpir/3.0.0")
+            self.requires("mpir/3.0.0", transitive_headers=True, transitive_libs=True)
```
Comment (prince-chrismc) [inline recipes/bliss/all/conanfile.py:49]: This is not the intended usage

```suggestion
            self.requires("mpir/3.0.0", transitive_headers=True)
```

File: recipes/bliss/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/bliss/all/conanfile.py:47]: Can you prove they are required? 

Moving forward the policy is going to be 

- if you dont know do not expose them
- if they are required put a comment

This way we can track and measure to test the hypothesis

**Review by SSE4 — APPROVED** [review]

**Pull request merged**